### DISCUSSION TITLE
Suggestions for printing arrays with record ToString

### DISCUSSION THREAD

**Discussion by bunnyi116** [dotnet/csharplang#7461] [General]
## Example：

```
Person person = new("Nancy", "Davolio")
{
    PhoneNumbers = new string[] { "123", "456" }
};

Console.WriteLine(person);

// Result
//  original: Person { FirstName = Nancy, LastName = Davolio, PhoneNumbers = System.String[] }
//  hope    : Person { FirstName = Nancy, LastName = Davolio, PhoneNumbers = [ 123, 456 ] }

record Person(string FirstName, string LastName)
{
    public string[] PhoneNumbers { get; init; } = Array.Empty<string>();
};

```

#### original
`Person { FirstName = Nancy, LastName = Davolio, PhoneNumbers = System.String[] }
`
#### hope
`Person { FirstName = Nancy, LastName = Davolio, PhoneNumbers = [ 123, 456 ] }`

## Suggestion

I want the compiler to print out the contents of the array in the ToString in the record class, so that you can see the contents of the array more intuitively

**Comment by CyrusNajmabadi** [comment]
@Bunnui by default the compiler always just defers to the specific type to control all these things (equality, hasing, printing, etc.)

If you want to customize this, you can override the appropriate members in your record to do so.

**Reply by colejohnson66** [reply]
To expand, I believe it would be possible for a source generator to fill in the gap here. It could detect an `[ArraySensitiveToString]` attribute and generate the (entire) custom `ToString` overload for you.
